terraform/modules/kubernetes/azure_postgres.tf

```diff
@@ -30,6 +30,40 @@ resource "azurerm_postgresql_flexible_server" "postgres-server" {
     ]
   }
 }
+resource "azurerm_postgresql_flexible_server" "postgres-snapshot-server" {
+  count = var.snapshot_databases_to_deploy ? 1 : 0
+
+  name                   = local.postgres_snapshot_server_name
+  location               = data.azurerm_resource_group.app-resource-group.location
+  resource_group_name    = data.azurerm_resource_group.app-resource-group.name
+  version                = var.postgres_version
+  administrator_login    = var.postgres_admin_username
+  administrator_password = var.postgres_admin_password
+  create_mode            = "Default"
+  storage_mb             = var.postgres_flexible_server_storage_mb
+  sku_name               = var.postgres_snapshot_flexible_server_sku
+  delegated_subnet_id    = data.azurerm_subnet.postgres-subnet[0].id
+  private_dns_zone_id    = data.azurerm_private_dns_zone.postgres-dns[0].id
+  lifecycle {
+    ignore_changes = [
+      tags,
+      # Allow Azure to manage deployment zone. Ignore changes.
+      zone,
+      # Required for import because of https://github.com/hashicorp/terraform-provider-azurerm/issues/15586
+      create_mode
+    ]
+  }
+}
+
+resource "azurerm_postgresql_flexible_server_database" "snapshot-db" {
+  count = var.snapshot_databases_to_deploy ? 1 : 0
+  # add option to not create database
+
+  name      = local.postgres_snapshot_service_name
+  server_id = azurerm_postgresql_flexible_server.postgres-snapshot-server[0].id
+  collation = "en_US.utf8"
+  charset   = "utf8"
+}
 
 resource "azurerm_postgresql_flexible_server_configuration" "postgres-extensions" {
```

Comment:
The configuration should be applied to the snapshot database as well

Reply:
Added the configuration